rules/platform/identity-provider-configuration.yaml

```diff
@@ -3,9 +3,11 @@
 kind: Rule
 checkType: Platform
 title: Verify at least one identity provider has been configured
-expression: cluster.spec.identityProviders.size() >= 1
+expression: var1.spec.identityProviders.size() >= 1
```

Comment:
Seems to work, but on this specific rule I get a panic.
```console
[alderaan cel-scanner-poc{Vincent056/main}]$ go run main.go -i rules/platform/identity-provider-configuration.yaml
Using /home/wsato/Documents/openshift/4.16/aws_cluster/auth/kubeconfig to establish connection with cluster
Fetching config.openshift.io/v1/oauths/cluster
Key: var1
Value: &{map[apiVersion:config.openshift.io/v1 kind:OAuth metadata:map[annotations:map[include.release.openshift.io/ibm-cloud-managed:true include.release.openshift.io/self-managed-high-availability:true release.openshift.io/create-only:true] creationTimestamp:2024-08-01T08:08:21Z generation:1 managedFields:[map[apiVersion:config.openshift.io/v1 fieldsType:FieldsV1 fieldsV1:map[f:metadata:map[f:annotations:map[.:map[] f:include.release.openshift.io/ibm-cloud-managed:map[] f:include.release.openshift.io/self-managed-high-availability:map[] f:release.openshift.io/create-only:map[]] f:ownerReferences:map[.:map[] k:{"uid":"8bf59c01-9b30-404c-9c5f-be640073c5d3"}:map[]]] f:spec:map[]] manager:cluster-version-operator operation:Update time:2024-08-01T08:08:21Z]] name:cluster ownerReferences:[map[apiVersion:config.openshift.io/v1 kind:ClusterVersion name:version uid:8bf59c01-9b30-404c-9c5f-be640073c5d3]] resourceVersion:1603 uid:0c184b16-5cb1-4832-8837-ededc2a2face] spec:map[]]}
panic: Failed to evaluate CEL expression: no such key: identityProvider
```

I've got a cluster from the installer it has an empty `spec`:
```yaml
[alderaan cel-scanner-poc{Vincent056/main}]$ oc get oauth cluster -oyaml
apiVersion: config.openshift.io/v1
kind: OAuth
metadata:
  annotations:
    include.release.openshift.io/ibm-cloud-managed: "true"
    include.release.openshift.io/self-managed-high-availability: "true"
    release.openshift.io/create-only: "true"
  creationTimestamp: "2024-08-01T08:08:21Z"
  generation: 1
  name: cluster
  ownerReferences:
  - apiVersion: config.openshift.io/v1
    kind: ClusterVersion
    name: version
    uid: 8bf59c01-9b30-404c-9c5f-be640073c5d3
  resourceVersion: "1603"
  uid: 0c184b16-5cb1-4832-8837-ededc2a2face
spec: {}
```